CenterForOpenScience/ember-osf-web
# [EMB-347] Embosfed redesigned sign up page

## Purpose

Embosf the sign-up page.

## Summary of Changes

### Added
- Routes:
    - `register` - sign up page
- Utils:
    - `param` - drop-in replacement for jQuery.param
- Tests:
    - `register` route acceptance test
    - `param` util unit test
### Changed
- Components:
    - `validated-input/recaptcha` - added ability to bind action to reset recaptcha
    - `sign-up-form` - added submit task & user-registration model creation
    - `osf-navbar/auth-dropdown` - make Sign Up button transition to register route, if enabled
- Routes:
    - `home` - remove submit task & user-registration model creation (moved to `sign-up-form` component)
- Tests:
    - Removed captcha visibility assertions from logged-out home page test

## Side Effects

None known.

## Feature Flags

`ember_auth_register`

## QA Notes

This change requires QA of the sign-up page (`/register`) and the sign-up button in the nav bar. When the `ember_auth_register` flag is enabled, `/register` should serve the new Embosfed sign up page and the Sign Up button on Embosfed pages should take the user to this page without a page reload. QA should test the Sign Up button from several pages both legacy and Embosfed. When signing up via an institution, the user should be redirected back to the page on which they clicked the Sign Up button. The sign up form on the logged out home page should also be tested as the sign up page uses the same component and modifies it a bit. Cross-browser testing is recommended. I would expect the level of risk to be fairly low, but this *is* our sign-up page so a failure could have a large impact.

## Ticket

https://openscience.atlassian.net/browse/EMB-347

# Reviewer Checklist
